anta/tests/hardware.py

```diff
@@ -379,3 +379,58 @@ def test(self) -> None:
         # Verify that the expected redundancy protocol configured, operational and switchover ready
         elif not command_output["switchoverReady"]:
             self.result.is_failure(f"Redundancy protocol switchover status mismatch - Expected: True Actual: {command_output['switchoverReady']}")
+
+
+class VerifyTridentCounters(AntaTest):
+    """Verifies the trident counters for all interfaces.
+
+    Compatible with EOS operating in `trident` platform.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if all interfaces have drop and error counter values below the drop threshold.
+    * Failure: The test will fail if any interface has drop or error counter values above the drop threshold.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.hardware:
+      - VerifyTridentCounters:
+    ```
+    """
+
+    categories: ClassVar[list[str]] = ["hardware"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show platform trident counters", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyTridentCounters test."""
+
+        drop_threshold: PositiveInteger = 0
+        """The upper limit for the configured packet drops or error counters."""
```

Comment:
Following the other tests:
```suggestion
        packet_drop_threshold: PositiveInteger = 0
        """Threshold for the number of dropped packets."""
```